### PR TITLE
Probably fixed shitty handling of Warp Tumours.

### PR DESCRIPTION
**Pull request by draknyte1**
Tumour handling was being fired in LivingHurtEvent, but now also fires in LivingDeathEvent.
Fixes https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/7093.

### CODE REVIEW

File: src/main/java/com/kentington/thaumichorizons/common/lib/EventHandlerEntity.java [hunk 1]
```diff
@@ -531,6 +531,17 @@ public void livingDeath(final LivingDeathEvent event) {
...
             }
+            else if (event.entityLiving instanceof EntityPlayer) {
+            	final EntityInfusionProperties prop = (EntityInfusionProperties)event.entity.getExtendedProperties("CreatureInfusion");
+            	if (prop != null) {
+            		if (prop.tumorWarpPermanent > 0 || prop.tumorWarp > 0 || prop.tumorWarpTemp > 0) {
+            			Thaumcraft.proxy.getPlayerKnowledge().addWarpPerm(event.entity.getCommandSenderName(), prop.tumorWarpPermanent);
+            			Thaumcraft.proxy.getPlayerKnowledge().addWarpSticky(event.entity.getCommandSenderName(), prop.tumorWarp);
+            			Thaumcraft.proxy.getPlayerKnowledge().addWarpTemp(event.entity.getCommandSenderName(), prop.tumorWarpTemp);
+            		}
+            		prop.resetPlayerInfusions();
```
Comment (TheElan) [inline src/main/java/com/kentington/thaumichorizons/common/lib/EventHandlerEntity.java:542]: just curious, why do we want to remove player infusions in any case?

**Comment by TheElan** [comment]
I don't think this will fix the issue 100%. From what I see in changes you should gain back the warp you lost using tumor, but I guess this still doesn't solve the issue with gaining new warp after using the tumor.

Was this feature working before at all?
Maybe there is also something broken in the code that activates the tumor?

**Comment by TheElan** [comment]
if this was tested and at least returns warp back, I would say - let's merge it and fix new warp gaining in separate PR, wdyt?

**Comment by draknyte1** [comment]
> I don't think this will fix the issue 100% (yeah, read the "probably" in title). From what I see in changes you should gain back the warp you lost using tumor, but I guess this still doesn't solve the issue with gaining new warp after using the tumor.
> 
> Was this feature working before at all? Maybe there is also something broken in the code that activates the tumor?

New warp is still gained normally after getting tumours and isn’t removed. So it’s handled fine. 

7D0E1000-02BD-44FF-9010-7993B3F49E62

I just moved the original code to the correct event handler. I assume all infusions should be removed on death? Idk, I don’t thaumcraft. To my knowledge it only didn’t work because of how it was handled for the reapply, cause it never would’ve worked on the damage event. Should now work as intended on death though. (Unless thaumcraft requires the player receiving warp to be alive?) I didn’t test it because I don’t know how, I don’t use TC, let alone player infusion. Someone who knows how should test it to confirm.

File: src/main/java/com/kentington/thaumichorizons/common/lib/EventHandlerEntity.java (hunk 1 above)
Comment (draknyte1) [reply src/main/java/com/kentington/thaumichorizons/common/lib/EventHandlerEntity.java:542]: That’s just what the default code did, dunno. 🤷🏻‍♀️

**Comment by TheElan** [comment]
If that's original code - then it should be fine. I would suggest pinging issue author to try to reproduce it with new version, just in case they have some wired edge case.

**Comment by bombcar** [comment]
yay it's merged let me try to build and test, worst case I'll report a bug against 2.1.1.5

**Comment by bombcar** [comment]
Confirmed that tumor now works as expected, restores warp on death in SP, and does NOT prevent gaining warp while using it. 

HOWEVER - I would need to really test in "non-cheats" mode and 100% verify that warp from normal thaum activities applies. Tested by eating brains and using /tc warp add

**Comment by bombcar** [comment]
This fixes https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/5404 but maybe also fixes https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/7093
